src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/storage/ContainerCleanerIntTest.java

```diff
@@ -0,0 +1,193 @@
+
+    @Autowired
+    private EnvelopeRepositoryImpl envelopeRepository;
+
+    @Autowired
+    private DbHelper dbHelper;
+
+    @BeforeAll
+    static void setUpTestMode() {
+        StorageClientsHelper.setAzureTestMode();
+        setupClass();
+
+        dockerComposeContainer =
+            new DockerComposeContainer(new File("src/integrationTest/resources/docker-compose.yml"))
+                .withExposedService("azure-storage", 10000);
+        dockerComposeContainer.start();
+    }
+
+    @AfterAll
+    static void tearDownTestMode() {
+        dockerComposeContainer.stop();
+    }
+
+    @BeforeEach
+    void setUp() {
+        dbHelper.deleteAll();
+
+        storageClient =
+            new BlobServiceClientBuilder()
+                .connectionString("UseDevelopmentStorage=true")
+                .buildClient();
+        containerClient = storageClient.createBlobContainer(CONTAINER_NAME);
+
+        containerCleaner = new ContainerCleaner(storageClient, envelopeRepository);
+    }
+
+    @AfterEach
+    void tearDown() {
+        storageClient.deleteBlobContainer(CONTAINER_NAME);
+    }
+
+    @Test
+    public void should_find_blobs_and_delete() {
+        // given
+        String[] dispatchedFileNames = new String[]{TEST_1, TEST_3};
+
+        uploadFilesToStorage(dispatchedFileNames);
+
+        // ensure files have been uploaded
+        assertThat(containerClient.listBlobs()).hasSize(dispatchedFileNames.length);
+        assertThat(containerClient.listBlobs().stream().map(BlobItem::getName))
+            .containsExactlyInAnyOrder(dispatchedFileNames);
+
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+
+        // when
+        containerCleaner.process(CONTAINER_NAME);
+
+        // then
+        assertThat(containerClient.listBlobs()).isEmpty();
+        assertFilesIsDeleteState(true, dispatchedFileNames);
+    }
+
+    @Test
+    public void should_delete_only_dispatched_blobs() {
+        // given
+        String[] dispatchedFileNames = new String[]{TEST_1, TEST_3};
+        String[] rejectedFileNames = new String[]{TEST_2, TEST_4};
+
+        uploadFilesToStorage(dispatchedFileNames);
+        uploadFilesToStorage(rejectedFileNames);
+
+        // ensure files have been uploaded
+        assertThat(containerClient.listBlobs()).hasSize(dispatchedFileNames.length + rejectedFileNames.length);
+        assertThat(containerClient.listBlobs().stream().map(BlobItem::getName))
+            .containsExactlyInAnyOrder((String[]) addAll(dispatchedFileNames, rejectedFileNames));
+
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+        createEnvelopes(REJECTED, rejectedFileNames);
+
+        // when
+        containerCleaner.process(CONTAINER_NAME);
+
+        // then
+        assertThat(containerClient.listBlobs()).hasSize(rejectedFileNames.length);
+        assertThat(containerClient.listBlobs().stream().map(BlobItem::getName))
+            .containsExactlyInAnyOrder(rejectedFileNames);
```

Comment:
the second assertions covers the first one so the first one can be deleted.

Reply:
Yes, indeed